source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -216,7 +216,8 @@ int DefaultCertValidator::doVerifyCertChain(
   }
 
   return allow_untrusted_certificate_ ? 1
-                                      : (validated != Envoy::Ssl::ClientValidationStatus::Failed);
+         : verify_trusted_ca_         ? validated != Envoy::Ssl::ClientValidationStatus::Failed
```

Comment:
style: please don't nest `? :`, it's hard to read

Reply:
Thanks for the review! Acked and updated.